### ISSUE TITLE
508-defect-2 Fast Follow! [COGNITION]:  Downloaded files SHOULD be consistent with what the user clicked to download

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12184]
# [508-defect-2 Fast Follow!](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Fast follow means it's the first thing to be worked on after launch
```

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

For the "Download letter (PDF)" link,  we need to ensure that the downloaded file has a unique name that can be understood by the user. Currently, each download is named "letter.pdf" and additional files are appended with "(1)", etc. Since the downloaded filename doesn't resemble the information the user clicked they may not be able to locate the file. Might we change the Content-Disposition HTTP header's filename parameter so that downloads will use the `download` attribute value, and the filename will align with what the user expects they are downloading?

For the stressful scenario of a person downloading their debt letters, it **should** be helpful if they can distinguish between the individual debt letters, and a series of "letter.pdf," "letter(1).pdf," and so on would only compound their stress. 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a user, I want to know what document I am downloading is and recognize the file I downloaded.

## Environment

* Operating System: all
* Browser: all
* Server destination: staging

## Steps to Recreate

1. Sign in as user 1, enter https://staging.va.gov/manage-va-debt/your-debt/ in browser
2. Navigate to the bottom section with the Download letter (PDF) links
3. Download a letter, verify that it downloads as letter.pdf
4. Download a different letter, verify that it downloads as letter (1).pdf

## Potential fix

1. Remove the `download` attribute
1. On the backend, set the file name to follow the pattern: `"typeDescription"` + "received" + `"receivedAt"` 
1. Work with the designers and Peggy to communicate to the user that it may take a minute or more to download the letter
1. Add `type="application/pdf"` to the link

[Note: adding `target="_blank"` to the download link helped to signal to the user something was happening, addressed the previous delay. Nice!]

## WCAG or Vendor Guidance (optional)

* [David Walsh - the download attribute](https://davidwalsh.name/download-attribute)

* [Chrome Dev Resources - Downloading resources in HTML5: a[download]](https://developers.google.com/web/updates/2011/08/Downloading-resources-in-HTML5-a-download)

* [MDN Content-Disposition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition)

## Screenshots or Trace Logs

![Screen Shot 2020-08-06 at 12 09 54 PM](https://user-images.githubusercontent.com/57469/89555245-cb7e0780-d7dd-11ea-942e-6f1e26057d45.png)

![Screen Shot 2020-08-06 at 12 18 45 PM](https://user-images.githubusercontent.com/57469/89556288-22381100-d7df-11ea-859f-cd7ad0c6ab65.png)

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

**Comment by bradl3yC** [comment]
@jenstrickland as per our slack discussion - we do not have file names accessible to us.  It is an API call.  We only have a file ID which does not match the file download.  Every single download link should result in a letter.pdf.  You saw letter(1) because you had already downloaded a letter and the (1) is appended by your operating system.

**Comment by jenstrickland** [comment]
@bradl3yC I hear ya. The download attribute can change the filename of the PDF. If the platform team is able to adjust the Content-Disposition HTTP header's filename parameter, we can make this happen. That's why [I reached out to VSP to see if they can help with this](https://dsva.slack.com/archives/CBU0KDSB1/p1596742928177700). There is nothing your team can do on this.

**Comment by jenstrickland** [comment]
According to VSP, they believe the backend team for this form/app have control of the headers returned by this service and the FE have control over the url used. This might be possible to improve the file download experience. https://dsva.slack.com/archives/CBU0KDSB1/p1596751922192300?thread_ts=1596742928.177700&cid=CBU0KDSB1

**Comment by bradl3yC** [comment]
This has been resolved by our BE engineer, Cameron

**Comment by jenstrickland** [comment]
I'm not able to validate this, as the url returns a 404.

The accessibility specialist should validate and close tickets.

**Comment by bradl3yC** [comment]
@jenstrickland I have turne debt letters back on in staging, please validate at your leisure

**Comment by jenstrickland** [comment]
Thanks, @bradl3yC . I tried to validate, but the letters aren't available. 

![Screen Shot 2020-09-23 at 5 43 40 PM](https://user-images.githubusercontent.com/57469/94072795-5f268a00-fdc4-11ea-9335-981ae1953665.png)

**Comment by leahkeeler** [comment]
@jenstrickland - Can this be closed? You can validate with user 38 Thanks!!!!!

**Comment by jenstrickland** [comment]
@leahkeeler Thanks so much for the info on which account may be used to validate the fix! There are a few things to address.

![Screen Shot 2020-10-28 at 10 57 16 AM](https://user-images.githubusercontent.com/57469/97453744-69303100-190c-11eb-9d0e-584bfde876a7.png)

1. The date displayed should match the file downloaded, to correspond with the user expectation. Perhaps this is only a test user issue in this instance, but I don't know if that's the case, so want to be sure we provide the date displayed with the file downloaded.

2. The download icon is about half the size of the icon used elsewhere (more info: [VA design system, icons](https://design.va.gov/design/icons)). For a correct implementation, [review this caregiver page](https://staging.va.gov/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/introduction). If y'all prefer, I can document this one in a new issue ticket.

3. The file type downloaded should be a PDF, not a TextEdit Document, to match what the user expects. This can be addressed by using the download link pattern, and including the type attribute, as shown here:

**Current code**

```html
<a target="_blank" rel="noopener noreferrer" 
    download="Debt Increase Letter dated May 28, 2020" 
    href="https://staging-api.va.gov/v0/debt_letters/%7B93631483-E9F9-44AA-BB55-3552376400D8%7D">
      <i aria-hidden="true" role="img" class="fas fa-download vads-u-padding-right--1"></i>
      <span aria-hidden="true">Download letter </span>
      <span class="sr-only">Download Second Demand Letter dated May 28, 2020</span>
      <dfn><abbr title="Portable Document Format">(PDF)</abbr></dfn>
</a>
```

**Recommended, to follow the download link pattern**

```diff
<a target="_blank" rel="noopener noreferrer" 
+   type="application/pdf"
    download="Debt Increase Letter dated May 28, 2020" 
    href="https://staging-api.va.gov/v0/debt_letters/%7B93631483-E9F9-44AA-BB55-3552376400D8%7D">
      <i aria-hidden="true" role="img" class="fas fa-download vads-u-padding-right--1"></i>
      <span aria-hidden="true">Download letter </span>
      <span class="sr-only">Download Second Demand Letter dated May 28, 2020</span>
      <dfn><abbr title="Portable Document Format">(PDF)</abbr></dfn>
</a>
```

**Comment by jenstrickland** [comment]
@bradl3yC 

See the download pane in the screenshot enclosed, it did not match what was expected. The date is wrong, and the file type isn't a PDF.

**Comment by bradl3yC** [comment]
@jenstrickland im confirming with @camerontesterman on the date in the file name.  The date in the letter itself matches the markup.  Also, I don't think we have a lot of control over the file type as this is a direct file request from VBMS.  Adding a type of pdf will not explicitly convert the file type.

**Comment by jenstrickland** [comment]
@bradl3yC Since your downloads are managed in the header (@camerontesterman 's realm), it's likely a setting there. For the VA design system download PDF link pattern, it does include the type attribute, so we should be consistent in usage. The file type in the link matters for assistive technology.

**Comment by bradl3yC** [comment]
@camerontesterman did we look into this? The dates in the file name aren't aligning with the dates that are in the data
